### PR TITLE
[protobuf] add support for mingw subsystem

### PR DESCRIPTION
**Pull request by fairlyblank**
**Describe the pull request**

- #### What does your PR fix?  
  protobuf build for mingw subsystem. no issue.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/protobuf/portfile.cmake [hunk 1]
```diff
@@ -78,6 +78,8 @@ protobuf_try_remove_recurse_wait(${CURRENT_PACKAGES_DIR}/debug/share)
 if(protobuf_BUILD_PROTOC_BINARIES)
     if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore" OR NOT VCPKG_CMAKE_SYSTEM_NAME)
         vcpkg_copy_tools(TOOL_NAMES protoc AUTO_CLEAN)
+    elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "MinGW")
+        vcpkg_copy_tools(TOOL_NAMES protoc AUTO_CLEAN)
```
Comment (dg0yt) [inline ports/protobuf/portfile.cmake:82]: Couldn't the condition be changed to `VCPKG_TARGET_IS_WINDOWS`, covering all flavours of Windows?

Comment (fairlyblank) [reply ports/protobuf/portfile.cmake:82]: yes, that's much more clear.

**Comment by PhoebeHui** [comment]
@fairlyblank, could you update the baseline version via  './vcpkg x-add-version --overwrite-version protobuf' and submit the changes again?

**Comment by fairlyblank** [comment]
> @fairlyblank, could you update the baseline version via './vcpkg x-add-version --overwrite-version protobuf' and submit the changes again?

I think i have updated the git-tree in last commit, after 'git rev-parse HEAD:ports/protubuf' .
I have run './vcpkg x-add-version --overwrite-version protobuf', but nothing need to commit.
Something i'm doing wrong here?

**Comment by PhoebeHui** [comment]
@fairlyblank, you didn't update the port-version in vcpkg.json, so it failed, I have updated it and committed the changes.

**Comment by fairlyblank** [comment]
> @fairlyblank, you didn't update the port-version in vcpkg.json, so it failed, I have updated it and committed the changes.

I missed that, thanks for your time.

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**